cli/src/description_util.rs

```diff
@@ -435,6 +429,21 @@ pub fn description_template(
     Ok(output.into_string_lossy())
 }
 
+pub fn print_default_description_deprecation_warning(
```

Comment:
You can inject warning via `default_config_migrations()`. See how `git.subprocess` is tested.

Reply:
Why would we rather use a `ConfigMigrationRule` for this? `git.subprocess` only uses one for when `git.subprocess` = `true` (and it doesn't look like that case is *tested* (e.g. in `tests/`) anywhere, but maybe I'm just misunderstanding), but uses a function like I did here when checking if the deprecation warning should be fired.

Reply:
I did it that way because someone setting `git.subprocess = false` might not immediately be able to move to the default (because of issues we haven’t yet fixed), so warning on every command would be spammy. If there’s a new configuration option users can immediately migrate to, just using the migration rule system is simpler.

Reply:
There's not one single option that can be moved to, though -- two options take the place of the one that people may have been using for either or both of the cases that the new options cover.

Reply:
Oh, you mean just warn, not automatically migrate. Gotcha.